packages/assets-controllers/src/DeFiPositionsController/fetch-positions.ts

```diff
@@ -0,0 +1,83 @@
+export type DefiPositionResponse = AdapterResponse<{
+  tokens: ProtocolToken[];
+}>;
+
+type ProtocolDetails = {
+  chainId: number;
+  protocolId: string;
+  productId: string;
+  name: string;
+  description: string;
+  iconUrl: string;
+  siteUrl: string;
+  positionType: PositionType;
+  metadata?: {
+    groupPositions?: boolean;
+  };
+};
+
+type AdapterResponse<ProtocolResponse> =
+  | (ProtocolDetails & {
+      chainName: string;
+    } & (
+        | (ProtocolResponse & { success: true })
+        | (AdapterErrorResponse & { success: false })
+      ))
+  | (AdapterErrorResponse & { success: false });
+
+type AdapterErrorResponse = {
+  error: {
+    message: string;
+  };
+};
+
+export type PositionType = 'supply' | 'borrow' | 'stake' | 'reward';
+
+export type ProtocolToken = Balance & {
+  type: 'protocol';
+  tokenId?: string;
+};
+
+export type Underlying = Balance & {
+  type: 'underlying' | 'underlying-claimable';
+  iconUrl: string;
+};
+
+export type Balance = {
+  address: string;
+  name: string;
+  symbol: string;
+  decimals: number;
+  balanceRaw: string;
+  balance: number;
+  price?: number;
+  tokens?: Underlying[];
+};
+
+// TODO: Update with prod API URL when available
+export const DEFAULT_DEFI_POSITIONS_API_URL =
```

Comment:
We still only have the dev url, no prod yet?

Reply:
Correct